### ISSUE TITLE
fix bug where cannot draw strand bound to other strand

### ISSUE THREAD

**Issue by dave-doty** [UC-Davis-molecular-computing/scadnano#883]
Create a strand in the forward direction:

![image](https://github.com/UC-Davis-molecular-computing/scadnano/assets/19274365/082a3eeb-f047-4df4-8ad9-60884dccb205)

In Pencil mode, try to draw a reverse strand within the same offsets. The strand will appear, but dragging the mouse while the button is down will not expand the strand:

![image](https://github.com/UC-Davis-molecular-computing/scadnano/assets/19274365/7b906006-5cce-414a-aca0-902500e726a9)

The same happens if a reverse strand is already present and you try to draw an overlapping forward strand:

![image](https://github.com/UC-Davis-molecular-computing/scadnano/assets/19274365/c85fb5b7-707c-48f9-8e5b-73d531920032)

**Comment by dave-doty** [comment]
I noticed that if you start drawing a strand outside the boundaries of another, it stops you from dragging it over the existing strand (despite one forward and the other reverse):

![image](https://github.com/UC-Davis-molecular-computing/scadnano/assets/19274365/b745e6d6-a263-4473-98eb-a80c227b389b)

So it seems likely that the bug is some bounds-checking code that is not properly accounting for forward/reverse booleans in domains.

**Comment by dave-doty** [comment]
I fixed this by reverting the code in state/design.dart, in the method `Deisng.domain_on_helix_at`. It's currently

```dart
  /// Return [Domain] at [address], INCLUSIVE, or null if there is none.
  Domain domain_on_helix_at(Address address, [StrandCreation strand_creation = null]) {
    for (var domain in this.helix_idx_to_domains[address.helix_idx]) {
      if (domain.contains_offset(address.offset) && domain.forward == address.forward) {
        return domain;
      }
    }
    return null;
  }
```

But to fix the bug in #855, it was changed to this:

```dart
  /// Return [Domain] at [address], INCLUSIVE, or null if there is none.
  Domain domain_on_helix_at(Address address, [StrandCreation strand_creation = null]) {
    for (var domain in this.helix_idx_to_domains[address.helix_idx]) {
      if (domain.contains_offset(address.offset) && domain.forward == address.forward) {
        return domain;
      } else if (strand_creation != null && overlap(domain, address.offset, strand_creation.start)) {
        return domain;
      }
    }
    return null;
  }
```

I'm not sure how to reproduce the bug in #855, but I'm putting this comment here for help with fixing that bug.

**Comment by dave-doty** [comment]
In case we re-open this issue, see this comment for why I reverted the change for it: https://github.com/UC-Davis-molecular-computing/scadnano/issues/883#issuecomment-1646740743
